### PR TITLE
Wrapper sockect functions

### PR DESCRIPTION
**Pull request by josgilmo**

### CODE REVIEW

**Comment by repejota** [comment]
Nice work. 
Could you rebase all those old merge comments before I merge everything please? :hammer: :smile_cat:

File: src/Connection.php [hunk 1]
```diff
@@ -99,19 +99,31 @@ public function getSubscriptions()
...
+
+    /**
      * Constructor.
      *
-     * @param ConnectionOptions $options Connection options object.
+     * @param ConnectionOptions $options       Connection options object.
+     * @param StreamWrapper     $streamWrapper Wrapper for stream functions. Testing purposes.
      */
-    public function __construct(ConnectionOptions $options = null)
+    public function __construct(ConnectionOptions $options = null, StreamWrapper $streamWrapper = null)
```
Comment (repejota) [inline src/Connection.php]: Could be possible to not use an $streamWrapper on the constructor?
Just a getter/setter would be fine.

**Comment by josgilmo** [comment]
Rebased :).
I think that it's ok now

**Comment by repejota** [comment]
Thanks mate! :smile:

**Pull request merged**